src/net_processing.cpp

```diff
@@ -2899,15 +3014,30 @@ void ProcessMessage(
                 LogPrint(BCLog::MEMPOOL, "not keeping orphan with rejected parents %s\n",tx.GetHash().ToString());
                 // We will continue to reject this tx since it has rejected
                 // parents so avoid re-requesting it from other peers.
+                // Here we add both the txid and the wtxid, as we know that
+                // regardless of what witness is provided, we will not accept
+                // this, so we don't need to allow for redownload of this txid
+                // from any of our non-wtxidrelay peers.
                 recentRejects->insert(tx.GetHash());
+                recentRejects->insert(tx.GetWitnessHash());
             }
         } else {
-            if (!tx.HasWitness() && state.GetResult() != TxValidationResult::TX_WITNESS_MUTATED) {
-                // Do not use rejection cache for witness transactions or
-                // witness-stripped transactions, as they can have been malleated.
-                // See https://github.com/bitcoin/bitcoin/issues/8279 for details.
+            if (state.GetResult() != TxValidationResult::TX_WITNESS_STRIPPED) {
```

Comment:
The `else if (tx.HasWitness() && RecursiveDynamicUsage(*ptx) < 100000)` block below is now dead code and should be removed (anything that doesn't go in this if branch is `TX_WITNESS_STRIPPED` and therefore doesn't have a witness)

Reply:
addressed in: https://github.com/bitcoin/bitcoin/pull/19879/commits/125c0381266e0e05a408f8e1818501ab73d29110, #19879